### PR TITLE
DOC: Fix typos and improve parts of docs

### PR DESCRIPTION
**Pull request by eu42**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: doc/source/getting_started/intro_tutorials/08_combine_dataframes.rst [hunk 1]
```diff
@@ -305,7 +305,7 @@ More information on join/merge of tables is provided in the user guide section o
     <div class="shadow gs-callout gs-callout-remember">
         <h4>REMEMBER</h4>
 
--  Multiple tables can be concatenated both column as row wise using
+-  Multiple tables can be concatenated both column- and row-wise using
```
Comment (mroeschke) [inline doc/source/getting_started/intro_tutorials/08_combine_dataframes.rst:308]: Maybe `column-wise and row-wise`

File: doc/source/user_guide/basics.rst [hunk 2]
```diff
@@ -1022,7 +1022,7 @@ Mixed dtypes
 ++++++++++++
 
 When presented with mixed dtypes that cannot aggregate, ``.agg`` will only take the valid
-aggregations. This is similar to how groupby ``.agg`` works.
```
Comment (mroeschke) [inline doc/source/user_guide/basics.rst:1025]: This is correct I believe. I think it's referring to the `groupby.agg` functionality

Comment (eu42) [reply doc/source/user_guide/basics.rst:1025]: I see, I was not aware of that function.

File: doc/source/getting_started/intro_tutorials/08_combine_dataframes.rst (hunk 1 above)
Comment (eu42) [reply doc/source/getting_started/intro_tutorials/08_combine_dataframes.rst:308]: There was a similar wording in `getting_started/basics`, as `...row- or column-wise, or elementwise.`, I tried to be consistent. Your suggestion also seems suitable.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @eu42! Failing build was unrelated

**Pull request merged**